UnityProject/Assets/Scripts/Objects/Machines/Gibber.cs

```diff
@@ -0,0 +1,158 @@
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			if (interaction.HandObject != null)
+			{
+				storage.StoreObject(interaction.HandObject);
+				return;
+			}
+
+			isRunning = !isRunning;
+			if (isRunning) StartCoroutine(GibbingTime());
+		}
+
+		public bool WillInteract(MouseDrop interaction, NetworkSide side)
+		{
+			// Only interact with this if it's anchored.
+			if (physics.isNotPushable == false) return false;
+			//Checks if the player is in reach + if they are able to interact with this type of object.
+			return DefaultWillInteract.Default(interaction, side);
+		}
+
+		public void ServerPerformInteraction(MouseDrop interaction)
+		{
+			storage.StoreObject(interaction.DroppedObject);
+		}
+
+		private IEnumerator GibbingTime()
+		{
+			Chat.AddLocalMsgToChat("The gibber violently shakes as it shreds everything inside of it.", gameObject);
+			lights.SetSpriteSO(lightson);
+			int time = 0;
+			while (isRunning)
+			{
+				if(timeToGib <= time) break;
+				time++;
+				yield return WaitFor.Seconds(0.6f);
+				yield return CheckContentAndHarm();
+			}
+
+			storage.RetrieveObjects();
+			foreach (var products in gibbed)
+			{
+				_ = Spawn.ServerPrefab(products.Key, gameObject.TileWorldPosition().To3(), gameObject.RegisterTile().Matrix.transform,
+					null, Mathf.Min(1, products.Value));
+			}
+			gibbed.Clear();
+			isRunning = false;
+			Chat.AddLocalMsgToChat("The gibber stops vibrating as it finishes its operation.", gameObject);
+			lights.SetSpriteSO(lightsoff);
+		}
+
+		private IEnumerator CheckContentAndHarm()
+		{
+			foreach (var slot in storage.GetStoredObjects().Reverse())
+			{
+				if (slot.TryGetComponent<LivingHealthMasterBase>(out var gib))
+				{
+					gib.ApplyDamageAll(gameObject, gib.PainScreamDamage + damagePerFrame,
+						AttackType.Melee, DamageType.Brute, true);
+					if (gib.OverallHealth > -100) continue;
+					var meatToProduce = gib.MeatProduce != null ? gib.MeatProduce : defaultProduce;
+					var skinToProduce = gib.SkinProduce != null ? gib.SkinProduce : defaultProduce;
+					AddItemsThatWillBeSpawned(meatToProduce, skinToProduce);
+					storage.RemoveObject(slot);
+					yield return WaitFor.EndOfFrame;
+					gib.OnGib();
+					continue;
+				}
+
+				if (slot.TryGetComponent<LivingHealthBehaviour>(out var oldMob))
+				{
+					oldMob.Death();
+					AddItemsThatWillBeSpawned(defaultProduce, defaultProduce);
+					continue;
+				}
+				if (slot.TryGetComponent<Integrity>(out var integrity) == false) continue;
+				if (integrity.gameObject.Item() != null) continue;
+				//Non-meaty items shall be damaged.. in exchange of damaging the machine itself.
+				machineIntegrity.ApplyDamage(damagePerFrame / 2, AttackType.Melee, DamageType.Brute,
+					false, false, false, true);
+				integrity.ApplyDamage(damagePerFrame, AttackType.Melee, DamageType.Brute);
+			}
+		}
+
+		private void AddItemsThatWillBeSpawned(GameObject meat, GameObject skin)
```

Comment:
I think it would be better to have a method with one argument, that you call twice.

Reply:
That's quite nitpicky considering that this function does exactly what the intended behavior is in one line.

Reply:
The method body is the same thing twice, just with different variables. There's no reason to duplicate that code.

You could also refactor the duplicated code into another method, but I don't see the point: the current one is called in one place only.

Reply:
is this good

Reply:
That's not what I meant, sorry.

```csharp
private void AddItemThatWillBeSpawned(GameObject item)
```

Then use it like this:
```csharp
AddItemThatWillBeSpawned(meatToProduce);
AddItemThatWillBeSpawned(skinToProduce);
```